llama.cpp

```diff
@@ -224,35 +225,36 @@ enum llm_arch {
 };
 
 static const std::map<llm_arch, const char *> LLM_ARCH_NAMES = {
-    { LLM_ARCH_LLAMA,           "llama"      },
-    { LLM_ARCH_FALCON,          "falcon"     },
-    { LLM_ARCH_GROK,            "grok"       },
-    { LLM_ARCH_GPT2,            "gpt2"       },
-    { LLM_ARCH_GPTJ,            "gptj"       },
-    { LLM_ARCH_GPTNEOX,         "gptneox"    },
-    { LLM_ARCH_MPT,             "mpt"        },
-    { LLM_ARCH_BAICHUAN,        "baichuan"   },
-    { LLM_ARCH_STARCODER,       "starcoder"  },
-    { LLM_ARCH_PERSIMMON,       "persimmon"  },
-    { LLM_ARCH_REFACT,          "refact"     },
-    { LLM_ARCH_BERT,            "bert"       },
-    { LLM_ARCH_NOMIC_BERT,      "nomic-bert" },
-    { LLM_ARCH_BLOOM,           "bloom"      },
-    { LLM_ARCH_STABLELM,        "stablelm"   },
-    { LLM_ARCH_QWEN,            "qwen"       },
-    { LLM_ARCH_QWEN2,           "qwen2"      },
-    { LLM_ARCH_PHI2,            "phi2"       },
-    { LLM_ARCH_PLAMO,           "plamo"      },
-    { LLM_ARCH_CODESHELL,       "codeshell"  },
-    { LLM_ARCH_ORION,           "orion"      },
-    { LLM_ARCH_INTERNLM2,       "internlm2"  },
-    { LLM_ARCH_MINICPM,         "minicpm"    },
-    { LLM_ARCH_GEMMA,           "gemma"      },
-    { LLM_ARCH_STARCODER2,      "starcoder2" },
-    { LLM_ARCH_MAMBA,           "mamba"      },
-    { LLM_ARCH_XVERSE,          "xverse"     },
-    { LLM_ARCH_COMMAND_R,       "command-r"  },
-    { LLM_ARCH_UNKNOWN,         "(unknown)"  },
+    { LLM_ARCH_LLAMA,            "llama"      },
+    { LLM_ARCH_FALCON,           "falcon"     },
+    { LLM_ARCH_GROK,             "grok"       },
+    { LLM_ARCH_GPT2,             "gpt2"       },
+    { LLM_ARCH_GPTJ,             "gptj"       },
+    { LLM_ARCH_GPTNEOX,          "gptneox"    },
+    { LLM_ARCH_MPT,              "mpt"        },
+    { LLM_ARCH_BAICHUAN,         "baichuan"   },
+    { LLM_ARCH_STARCODER,        "starcoder"  },
+    { LLM_ARCH_PERSIMMON,        "persimmon"  },
+    { LLM_ARCH_REFACT,           "refact"     },
+    { LLM_ARCH_BERT,             "bert"       },
+    { LLM_ARCH_NOMIC_BERT,       "nomic-bert" },
+    { LLM_ARCH_BLOOM,            "bloom"      },
+    { LLM_ARCH_STABLELM,         "stablelm"   },
+    { LLM_ARCH_STABLELM2,        "stablelm2"   },
```

Comment:
nit, but curly brackets are misaligned

Reply:
Fixed !

Reply:
Also this is working, only caveat is you need to use the model from this branch of HF for now: https://huggingface.co/stabilityai/stablelm-2-12b/tree/stack-per-head-qk-norm . working to get it added to main branch soon.

Reply:
Tested `StableLM-2-12b, StableLM-2-12b-chat, StableLM-2-1_6b, StableLM-zephyr-3b`